lessons/misc/intro-bash/lesson.md

```diff
@@ -0,0 +1,354 @@
+```
+
+Hmmm...that didn't work. It turns out `rmdir` only works on empty directories. Let's make sure we don't actually need the files in that directory before trying again to delete it. You can pass directory names to `ls` to its contents.
+
+```
+ls folder-to-delete
+```
+
+Now that we're sure we don't need that folder anymore, let's **f**orcefully and **r**ecursively delete it using the `rm` (for *remove*) command. Note, this command will not warn you and the content will be gone forever.
+
+```
+rm -rf folder-to-delete
+```
+
+`rm` can also be used on files without the `-rf` flags. Try remove `to-delete.txt`
+
+## Working with files
+
+Beyond navigating directories, bash offers a suite of useful tools to do all sorts of things with files themselves.
+
+Let's examine `hello_world.py` in greater detail. We can do this with `cat`, which simply spits out the contents of a file to the terminal's output.
+
+```
+cat hello_world.py
+```
+
+When it comes to looking at individual files, however, `cat` is generally not very practical for anything but tiny files like this one. Generally, we would want to use `less`, a program known as a *pager* that is designed to let users scroll through files (no matter how large) by only generating a single 'page worth' at a time.
+
+Let's use `less` to view the contents of the larger file, `animals.txt`.
+
+```
+less animals.txt
+```
+
+Finally, bash features two useful tools that output the first ten lines (`head`) and last ten lines (`tail`) of a file. These are especially handy for large files that we don't want to open in their entirety.
+
+```
+head animals.txt
+tail animals.txt
+```
+
+## Copying and moving files
+
+`cp` is used to copy files. When using `cp`, you have to type out both the file you want to copy as well as the name of the file you want to make.
+
+```
+cp animals.txt animals_copy.txt
+```
+
+`cp` also allows you to copy into a folder by providing a folder name instead of a new file name.
+
+```
+cp animals.txt text-files/animals_copy.txt
+ls text-files # Confirm that file is there
+```
+
+`mv` (**m**o**v**e) is somewhat similar, but requires you to type out the file you want to move as well as the directory you want to move it to. Let's move our copy of `animals.txt` into `text_files`. Here, we'll also use `mv` to rename `animals_copy.txt` to `animals_copy_2.txt` upon placing it in `text-files`. Otherwise, the version of `animals_copy.txt` already in `text-files` would be overwritten. **Note:** Bash will not warn you before overwritting a file if a file with the same name already exists in the directory.
+
+```
+mv animals.txt text-files/
+ls text-files
+```
+
+Here, we'll also use `mv` to rename `animals_copy.txt` to `animals_copy_2.txt` and place it in `text-files`. Otherwise, the version of `animals_copy.txt` already in `text-files` would be overwritten. **Note:** Bash will not warn you before overwritting a file if a file with the same name already exists in the directory.
```

Comment:
Redundant with paragraph above - might want to delete the instance of this in the first paragraph before that chunk of code